src/vs/workbench/contrib/preferences/browser/settingsTree.ts

```diff
@@ -1035,25 +1035,30 @@ export class SettingObjectRenderer extends AbstractSettingRenderer implements IT
 			const defaultValue: Record<string, unknown> = template.context.defaultValue;
 			const scopeValue: Record<string, unknown> = template.context.scopeValue;
 			const newValue: Record<string, unknown> = {};
+			let newItems: IObjectDataItem[] = [];
 
-			template.objectWidget.items.forEach(item => {
+			template.objectWidget.items.forEach((item, idx) => {
 				// Item was updated
-				if (isDefined(e.item) && e.originalItem.key.data === item.key.data) {
+				if (isDefined(e.item) && e.targetIndex === idx) {
 					newValue[e.item.key.data] = e.item.value.data;
+					newItems.push(e.item);
 				}
-				// All remaining items
-				else {
+				// All remaining items, but skip the one that we just updated
+				else if (isUndefinedOrNull(e.item) || e.item.key.data !== item.key.data) {
```

Comment:
Consider we have files associations like this

|Item|Value|
|---|---|
|foo|bar|
|hello|world|

Let's say I change `foo` to `hello`

|Item|Value|
|---|---|
|hello|bar|
|hello|world|

Then the correct table should be:

|Item|Value|
|---|---|
|hello|bar|